eng/Migrate-Package.ps1

```diff
@@ -0,0 +1,204 @@
+$NewRepoRemoteName = "new-repo"
+
+# Go to origin repo
+Execute-Command "Set-Location $OriginRepoPath -Verbose"
+
+# Get the branch just in case
+Execute-Command "git fetch $OriginRemote $OriginBranch"
+
+# Switch to the desired branch
+Execute-Command "git checkout $OriginBranch"
+
+# Delete the existing branch, just in case
+Execute-Command "git branch -D $SplittedBranchName"
+
+# Filter commits relevant to assembly, may take a long time
+Execute-Command "git subtree split -P '$RelativeAssemblyPath' -b $SplittedBranchName"
+
+# Move to that branch
+Execute-Command "git checkout $SplittedBranchName"
+
+# Remove any files that don't belong
+Execute-Command "git reset --hard"
+
+# Put everything under the migrated directory
+Execute-Command "New-Item -ItemType Directory -Name '$MigratedFolderName' -Verbose"
+Execute-Command "Move-Item -Path '*' -Destination '$MigratedFolderName' -Exclude '$MigratedFolderName' -Verbose"
+
+Write-Color red "PAUSE"
+$IN = Read-Host "Please verify that the files that were moved to the '$MigratedFolderName' folder look good, then press any key to continue..."
+
+Execute-Command "git add *"
+Execute-Command "git commit -m 'Move everything to the migrated folder'"
+
+# Go back to maintenance-packages
+Execute-Command "Set-Location $DestinationRepoPath -Verbose"
+
+# Change to main and update it if needed
+Execute-Command "git checkout main"
+
+# Remove the remote if it already exists, just in case
+Execute-Command "git remote remove $NewRepoRemoteName"
+Execute-Command "git remote add $NewRepoRemoteName https://github.com/dotnet/maintenance-packages"
+Execute-Command "git fetch $NewRepoRemoteName main"
+Execute-Command "git checkout main"
+Execute-Command "git reset --hard $NewRepoRemoteName/main"
+
+# Delete the existing branch, just in case
+Execute-Command "git branch -D $WorkingBranchName"
+
+# Create the final branch
+Execute-Command "git checkout -b $WorkingBranchName"
+
+# Delete the existing branch, just in case
+Execute-Command "git branch -D $NoHistoryBranchName"
+
+# Create empty branch
+Execute-Command "git checkout --orphan $NoHistoryBranchName"
+
+# Remove all files that didn't go away, needs to be completely empty
+Execute-Command "git reset --hard"
+
+# Just in case, to avoid bringing in any leftover files
+Execute-Command "git clean -fdx"
+
+# Add origin repo as a remote
+Execute-Command "git remote add $OldRepoRemoteName $OriginRepoPath"
+
+# Bring in the filtered commits from old repo into maintenance-packages
+Execute-Command "git pull $OldRepoRemoteName $SplittedBranchName"
+
+Write-Color red "PAUSE"
+$IN = Read-Host "Please check the contents of $NoHistoryBranchName before merging with it into $WorkingBranchName, then press any key to continue..."
+
+# # Switch to final branch
+Execute-Command "git checkout $WorkingBranchName"
+
+# Bring in the changes that were migrated to the empty branch
+Execute-Command "git merge --allow-unrelated-histories $NoHistoryBranchName"
+
```

Comment:
I think I can add some more steps:

- Move the `migrated/` directory , initially located in the root directory, to the `src/` directory.
- Rename it to the name of the assembly we're moving.
- Delete the `*.sln` and `*.props` files.
- Delete the `pkg/` and `ref/` directories.
- Commit all that.

Reply:
On a second thought, I prefer not to, as the script is also meant to help migrate individual files or random directories, and those don't contain such files.